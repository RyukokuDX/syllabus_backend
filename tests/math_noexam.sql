-- 数理の科目の選択必修科目のうち、試験がないものの教科名と担当教員の一覧
-- File Version: v1.1.0
-- Project Version: v2.2.3
-- Last Updated: 2025-07-02

SELECT DISTINCT
	cache_data->>'科目名' as 教科名,
	instructor->>'氏名' as 担当教員
FROM syllabus_cache
	CROSS JOIN LATERAL (
		SELECT opening_info
		FROM jsonb_array_elements(cache_data->'開講情報一覧') as opening_info
		WHERE jsonb_typeof(opening_info) = 'object'
	) as opening
	CROSS JOIN LATERAL (
		SELECT syllabus
		FROM jsonb_array_elements(opening.opening_info->'シラバス一覧') as syllabus
		WHERE jsonb_typeof(syllabus) = 'object'
	) as syl
	CROSS JOIN LATERAL (
		SELECT instructor
		FROM jsonb_array_elements(syl.syllabus->'担当教員一覧') as instructor
		WHERE jsonb_typeof(instructor) = 'object'
	) as instructor
WHERE cache_name = 'subject_syllabus_cache'
  -- 履修要綱で数理の必修科目
  AND EXISTS (
    SELECT 1
    FROM jsonb_array_elements(cache_data->'履修情報一覧') as curriculum_info,
         jsonb_array_elements(curriculum_info->'履修要綱一覧') as requirement_info
    WHERE requirement_info->>'学部課程' LIKE '%数理%'
      AND requirement_info->>'必須度' = '必修'
  )
  -- シラバスで数理の学部課程に開講
  AND EXISTS (
    SELECT 1
    FROM jsonb_array_elements(syl.syllabus->'対象学部課程一覧') as faculty
    WHERE faculty::text LIKE '%数理%'
  )
  AND (
    -- 成績評価基準一覧が空の場合（試験がない）
    jsonb_array_length(syl.syllabus->'成績評価基準一覧') = 0
    OR
    -- 成績評価基準一覧に「試験」を含む項目がない場合
    NOT EXISTS (
      SELECT 1 
      FROM jsonb_array_elements(syl.syllabus->'成績評価基準一覧') as criteria
      WHERE criteria->>'項目' LIKE '%試験%'
    )
  )
  -- 数学系の科目（科目区分または科目小区分に「数学」または「数理」を含む）
  AND EXISTS (
    SELECT 1
    FROM jsonb_array_elements(cache_data->'履修情報一覧') as curriculum_info,
         jsonb_array_elements(curriculum_info->'履修要綱一覧') as requirement_info
    WHERE (
      requirement_info->>'科目区分' LIKE '%数学%'
      OR requirement_info->>'科目区分' LIKE '%数理%'
      OR requirement_info->>'科目小区分' LIKE '%数学%'
      OR requirement_info->>'科目小区分' LIKE '%数理%'
    )
  )
ORDER BY 教科名, 担当教員;