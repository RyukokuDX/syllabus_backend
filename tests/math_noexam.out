SQLファイルを実行中: tests/math_noexam.sql
          教科名           |  担当教員   
---------------------------+-------------
 セミナ-I                  | 角川 裕次
 セミナ-I                  | 高橋 隆史
 セミナ-I                  | 佐野 彰
 セミナ-I                  | 阪井 一繁
 セミナ-I                  | 山岸 義和
 セミナ-I                  | 松木平 淳太
 セミナ-I                  | 深尾 武史
 セミナ-I                  | 川上 竜樹
 セミナ-I                  | 村川 秀樹
 セミナ-I                  | 中野 浩
 セミナ-I                  | 藤原 和将
 セミナ-I                  | 馬 青
 セミナ-I                  | 飯田 晋司
 セミナ-I                  | 樋口 三郎
 セミナ-II                 | 角川 裕次
 セミナ-II                 | 高橋 隆史
 セミナ-II                 | 佐野 彰
 セミナ-II                 | 阪井 一繁
 セミナ-II                 | 山岸 義和
 セミナ-II                 | 深尾 武史
 セミナ-II                 | 川上 竜樹
 セミナ-II                 | 村川 秀樹
 セミナ-II                 | 中野 浩
 セミナ-II                 | 藤原 和将
 セミナ-II                 | 馬 青
 セミナ-II                 | 飯田 晋司
 セミナ-II                 | 樋口 三郎
 プログラミング及び実習I   | 中野 浩
 プロジェクト演習          | 角川 裕次
 プロジェクト演習          | 深尾 武史
 プロジェクト演習          | 村川 秀樹
 プロジェクト演習          | 藤原 和将
 プロジェクト演習          | 樋口 三郎
 情報基礎                  | 中野 浩
 数理·情報科学の学びと社会 | 阪井 一繁
 数理·情報科学の学びと社会 | 樋口 三郎
 数理情報演習              | 角川 裕次
 数理情報演習              | 高橋 隆史
 数理情報演習              | 阪井 一繁
 数理情報演習              | 山岸 義和
 数理情報演習              | 松木平 淳太
 数理情報演習              | 深尾 武史
 数理情報演習              | 川上 竜樹
 数理情報演習              | 村川 秀樹
 数理情報演習              | 中野 浩
 数理情報演習              | 藤原 和将
 数理情報演習              | 馬 青
 数理情報演習              | 飯田 晋司
 数理情報演習              | 樋口 三郎
 特別研究I                 | 角川 裕次
 特別研究I                 | 高橋 隆史
 特別研究I                 | 佐野 彰
 特別研究I                 | 阪井 一繁
 特別研究I                 | 山岸 義和
 特別研究I                 | 深尾 武史
 特別研究I                 | 川上 竜樹
 特別研究I                 | 村川 秀樹
 特別研究I                 | 中野 浩
 特別研究I                 | 藤原 和将
 特別研究I                 | 馬 青
 特別研究I                 | 飯田 晋司
 特別研究I                 | 樋口 三郎
 特別研究II                | 角川 裕次
 特別研究II                | 高橋 隆史
 特別研究II                | 佐野 彰
 特別研究II                | 阪井 一繁
 特別研究II                | 山岸 義和
 特別研究II                | 深尾 武史
 特別研究II                | 川上 竜樹
 特別研究II                | 村川 秀樹
 特別研究II                | 中野 浩
 特別研究II                | 藤原 和将
 特別研究II                | 馬 青
 特別研究II                | 飯田 晋司
 特別研究II                | 樋口 三郎
 微積分及び演習I           | 深尾 武史
(76 行)

